pkg/controllers/dlb/controller.go

```diff
@@ -93,8 +98,13 @@ func (c *controller) UpdateDaemonSet(rawObj client.Object, ds *apps.DaemonSet) (
 	}
 
 	if !reflect.DeepEqual(ds.Spec.Template.Spec.NodeSelector, dp.Spec.NodeSelector) {
-		ds.Spec.Template.Spec.NodeSelector = dp.Spec.NodeSelector
-		updated = true
+		if len(dp.Spec.NodeSelector) > 0 {
+			ds.Spec.Template.Spec.NodeSelector = dp.Spec.NodeSelector
+			updated = true
+		} else if !reflect.DeepEqual(ds.Spec.Template.Spec.NodeSelector, defaultNodeSelector) {
+			ds.Spec.Template.Spec.NodeSelector = defaultNodeSelector
+			updated = true
+		}
 	}
```

Comment:
I'd try to avoid extra call of reflect.DeepEqual as it's quite heavy.

How about this one?
```suggestion
	for _, ns := range []map[string]string{dp.Spec.NodeSelector, defaultNodeSelector} {
		if len(ns) > 0 && !reflect.DeepEqual(ds.Spec.Template.Spec.NodeSelector, ns) {
			ds.Spec.Template.Spec.NodeSelector = ns
			updated = true
			break
		}
	}
```

or this one?
```suggestion
	if len(dp.Spec.NodeSelector) > 0 {
		if !reflect.DeepEqual(ds.Spec.Template.Spec.NodeSelector, dp.Spec.NodeSelector) {
			ds.Spec.Template.Spec.NodeSelector = dp.Spec.NodeSelector
			updated = true
		}
	} else if !reflect.DeepEqual(ds.Spec.Template.Spec.NodeSelector, defaultNodeSelector) {
		ds.Spec.Template.Spec.NodeSelector = defaultNodeSelector
		updated = true
	}
```

Reply:
I see your point.! In my opinion, the second one looks more clear..! Thanks for the suggestion.! I feel like,, I am actually not writing any code...